linkerd/app/outbound/src/tcp/logical/router.rs

```diff
@@ -60,23 +61,91 @@ where
         // TODO(ver) use a different key type that is structured instead of
         // simply a name.
         let mut backends = HashMap::with_capacity(profile.targets.len().max(1));
-        for t in profile.targets.iter() {
+        Self::mk_backends(&mut backends, &self.0, &profile.targets, &target);
+
+        // Create a stack that can distribute requests to the backends.
+        let distribute = Self::mk_distribute(&backends, &profile.targets, &target);
+        let (tx, rx) = watch::channel(distribute.clone());
+        drop(profile);
+
+        let new_backend = self.0.clone();
+        let span = tracing::debug_span!("router", target = %Param::<profiles::LogicalAddr>::param(&target));
```

Comment:
it's fine to add a router span, but the target should already be on the "logical" parent span